platform/default/src/mbgl/storage/offline_database.cpp

```diff
@@ -606,6 +606,47 @@ bool OfflineDatabase::putTile(const Resource::TileData& tile,
     return true;
 }
 
+std::exception_ptr OfflineDatabase::invalidateTileCache() try {
+    // clang-format off
+    mapbox::sqlite::Query query{ getStatement(
+        "UPDATE tiles "
+        "SET expires = 0, must_revalidate = 1 "
+        "WHERE id NOT IN ("
+        "    SELECT tile_id FROM region_tiles"
+        ")"
+    ) };
```

Comment:
Did you conduct a benchmark and/or query analysis? I.e. can we execute it sufficiently fast for large databases?

Reply:
Good point, definitely deserves a benchamark test.

Reply:
AFAIK, this API is just for exceptional cases.. also comparing against cleaning-reloading the whole cache, this API should win anyway.

Reply:
Added benchmark tests. Interesting that cleaning a region is 5 times slower because of a string comparison.

Reply:
Done